config/externalnamenottested.go

```diff
@@ -751,4 +751,48 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
 	// aws_storagegateway_working_storage can be imported by using the gateway Amazon Resource Name (ARN) and local disk identifier separated with a colon (:)
 	// Example: arn:aws:storagegateway:us-east-1:123456789012:gateway/sgw-12345678:pci-0000:03:00.0-scsi-0:0:0:0
 	"aws_storagegateway_working_storage": config.TemplatedStringAsIdentifier("", "{{ .parameters.gateway_arn }}:{{ .parameters.disk_id }}"),
+
+	// networkfirewall
+	//
+	// Network Firewall Policies can be imported using their ARN
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:firewall-policy/example
+	"aws_networkfirewall_firewall_policy": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall-policy/{{ .external_name }}"),
+	// Network Firewall Logging Configurations can be imported using the firewall_arn
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:firewall/example
+	"aws_networkfirewall_logging_configuration": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall/{{ .external_name }}"),
+	// Network Firewall Resource Policies can be imported using the resource_arn
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:stateful-rulegroup/example
+	"aws_networkfirewall_resource_policy": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+	// Network Firewall Rule Groups can be imported using their ARN
+	// Example: arn:aws:network-firewall:us-west-1:123456789012:stateful-rulegroup/example
+	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+
+	// networkmanager
+	//
+	// aws_networkmanager_connection can be imported using the connection ARN
+	// Example: arn:aws:networkmanager::123456789012:device/global-network-0d47f6t230mz46dy4/connection-07f6fd08867abc123
+	"aws_networkmanager_connection": config.TemplatedStringAsIdentifier("", "arn:aws:networkmanager::{{ .setup.client_metadata.account_id }}:device/{{ .parameters.global_network_id }}/{{ .external_name }}"),
```

Comment:
I am not sure that, the last element of the template string will provide this string: `connection-07f6fd08867abc123`

I will suggest two things for this resource:

- We may consider using the existing configuration and let's see what is happening while testing.
- Or we may consider using IdentifierFromProvider here.

Decision is yours @dverveiko

Reply:
To better understanding of configuration process, I've decided to leave this configuration as is and check what will happen in testing. I'll leave a comment for that person who will test this.